node-util/sbin/oo-admin-cartridge

```diff
@@ -36,6 +36,11 @@ module OpenShift
                      [options.source]
                    end
 
+            if dirs.empty?
+              $stderr.puts "Provided directory is either empty or invalid."
+              return "Failed"
```

Comment:
This works and is a good change, but I think the wording can be improved. Take a look at the same failure when not providing the recursive flag:
```
Installation failed for /usr/libexec/openshift/cartridges/abc: Illegal path to cartridge source: '/usr/libexec/openshift/cartridges/abc'

Operation failed: some cartridges cannot be installed
```

We should be using similar wording. Returning just "Failed" is not very descriptive about the current state of the cartridge. "Operation failed: some cartridges cannot be installed" or something similar would be better.